src/rt/sections_elf_shared.d

```diff
@@ -711,12 +711,12 @@ void scanSegments(in ref dl_phdr_info info, DSO* pdso)
         case PT_LOAD:
             if (phdr.p_flags & PF_W) // writeable data segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._gcRanges.insertBack(beg[0 .. phdr.p_memsz]);
             }
             version (Shared) if (phdr.p_flags & PF_X) // code segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._codeSegments.insertBack(beg[0 .. phdr.p_memsz]);
```

Comment:
> Not all of it. Just why (p_vaddr & p_align) == (p_offset & p_align) implies !(dlpi_addr is unaligned && p_vaddr == 0).

I've just noticed that should be modulo (`%`), not bitwise and (`&`).  This is probably a sign I wrote that snippet as late at night as I'm reading right now.  Probably a sign to stop.

In any case, `p_align` is the alignment of `dlpi_addr`, and `p_align` should always be a power of 2.